builtin/credential/aws/path_config_rotate_root.go

```diff
@@ -0,0 +1,216 @@
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/vault/helper/awsutil"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func (b *backend) pathConfigRotateRoot() *framework.Path {
+	return &framework.Path{
+		Pattern: "config/rotate-root",
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathConfigRotateRootUpdate,
+			},
+		},
+
+		HelpSynopsis:    pathConfigRotateRootHelpSyn,
+		HelpDescription: pathConfigRotateRootHelpDesc,
+	}
+}
+
+func (b *backend) pathConfigRotateRootUpdate(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	// First get the AWS key and secret and validate that we _can_ rotate them.
+	// We need the read lock here to prevent anything else from mutating it while we're using it.
+	b.configMutex.RLock()
+	alreadyReleasedRLock := false
+	defer func() {
+		if alreadyReleasedRLock {
+			return
+		}
+		b.configMutex.RUnlock()
+	}()
+
+	clientConf, err := b.nonLockedClientConfigEntry(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if clientConf == nil {
+		return logical.ErrorResponse(`can't update client config because it's unset`), nil
+	}
+	if clientConf.AccessKey == "" {
+		return logical.ErrorResponse("can't update access_key because it's unset"), nil
+	}
+	if clientConf.SecretKey == "" {
+		return logical.ErrorResponse("can't update secret_key because it's unset"), nil
+	}
+
+	// Getting our client through the b.clientIAM method requires values retrieved through
+	// the user providing an ARN, which we don't have here, so let's just directly
+	// make what we need.
+	staticCreds := &credentials.StaticProvider{
+		Value: credentials.Value{
+			AccessKeyID:     clientConf.AccessKey,
+			SecretAccessKey: clientConf.SecretKey,
+		},
+	}
+	// By default, leave the iamEndpoint nil to tell AWS it's unset. However, if it is
+	// configured, populate the pointer.
+	var iamEndpoint *string
+	if clientConf.IAMEndpoint != "" {
+		iamEndpoint = aws.String(clientConf.IAMEndpoint)
+	}
+	awsConfig := &aws.Config{
+		Credentials: credentials.NewCredentials(staticCreds),
+		Endpoint:    iamEndpoint,
+
+		// Generally speaking, GetOrDefaultRegion will use the Vault server's region. However, if this
+		// needs to be overridden, an easy way would be to set the AWS_DEFAULT_REGION on the Vault server
+		// to the desired region. If that's still insufficient for someone's use case, in the future we
+		// could add the ability to specify the region either on the client config or as part of the
+		// inbound rotation call.
+		Region: aws.String(awsutil.GetOrDefaultRegion(b.Logger(), "")),
+
+		// Prevents races.
+		HTTPClient: cleanhttp.DefaultClient(),
+	}
+	sess, err := session.NewSession(awsConfig)
+	if err != nil {
+		return nil, err
+	}
+	iamClient := getIAMClient(sess)
```

Comment:
You might be able to get a client via `b.clientIAM` without having to deal with the aws config logic above (getIAMClient also seems to no longer exist).

Reply:
getIAMClient is implemented at the bottom of that file.

Reply:
I [don't see it](https://github.com/hashicorp/vault/blob/413c7edc0d76a6ffac16cba4197bb94c91e331ce/builtin/credential/aws/client.go#L248) on the master branch, can you merge origin/master into this branch?

Reply:
https://github.com/hashicorp/vault/pull/9921/files#diff-0da954d71882b5e1a019b3245078869bR211-R213